library/std/src/panicking.rs

```diff
@@ -256,7 +257,10 @@ fn default_hook(info: &PanicInfo<'_>) {
 
     let write = |err: &mut dyn crate::io::Write| {
         let _ = writeln!(err, "thread '{name}' panicked at '{msg}', {location}");
-
+        if let Some(source) = info.source() {
+            let report = Report::new(source).pretty(true);
+            let _ = writeln!(err, "Source: {report}");
```

Comment:
This'd more closely match [Jane's suggestion](https://github.com/rust-lang/rust/pull/103169/files#r997556789):
```suggestion
            let _ = writeln!(err, "\nSource: {report}\n");
```

(We can leave the exact output format for the default hook for later, but this is an easy edit to make it a bit nicer for now.)